# Epigenetic_plasticity_PAH_tolerance_killifish
Analysis for the project: Epigenetic plasticity is a driver of heritable pollution tolerance in killifish 

The repository contains the analysis code associated with the manuscript "Epigenetic plasticity is a driver of heritable pollution tolerance in Fundulus heteroclitus".

## Scope
The code provided illustrates the core analytical approaches used in htis study to evaluate how environmental PAH influences DNA methylation, SNP/SNV, RNA-seq, and cis-nat
expression in PAH-sensitive and PAH-tolerant Fundulus heteroclitus. Full bioinformatic pipelines (read processing, alignment and genome assembly) are not included. 

## Data availability
Raw sequencing data are available through NCBI GEO and GenBank under accessions provided below and in our manuscript: 

 Submission: SUB15839686 
 PRJNA1381666 : Fundulus heteroclitus Genome sequencing
 SAMN54117932 : Kings_Creek_Fundulus_heteroclitus

 ## Code Availability
 Analysis scripts will be continuously updated to this repository prior to publication

 ## Notes
 This repository is intended to support transparency and reproducibility of key results. Additional scirpts are available upon request (colwelma@ohsu.edu)

##### Software and Package Versions

All analyses were performed using the following software and R packages.

**Operating system**
- Linux (Ubuntu 22.04)

**Base software**
- R (v4.3.2)
- samtools (v1.18)
- bedtools (v2.31.0)
- STAR (v2.7.11b)
- StringTie (v2.2.1)
- modkit (v0.5.1)
- methylartist (v1.0.0)
- LiftOff (v1.6.3)

**R packages**
- methylKit (v1.30.0)
- GenomicRanges (v1.54.1)
- rtracklayer (v1.62.0)
- genomation (v1.36.0)
- ggplot2 (v3.5.0)
- dplyr (v1.1.4)
- tidyr (v1.3.0)
- readr (v2.1.5)
- UpSetR (v1.4.0)
- pheatmap (v1.0.12)
- data.table (v1.15.0)
- purrr (v1.0.2)


 ## Last updated: 12/16/25 MLC
